docs/manual/inference.tex

```diff
@@ -140,46 +142,86 @@
-\end{sloppypar}
+To run whole-program inference on a single project, use the \<wpi.sh>
+script (Section~\ref{wpi-one}).  To run it on many projects (say, when
+running it on projects from GitHub), use the \<wpi-many.sh> script
+(Section~\ref{wpi-one}).  All WPI-related scripts appear in the
+\<checker/bin/> directory.
 
-\begin{enumerate}
-\item Optional: Command-line arguments to
-  \href{https://checkerframework.org/annotation-file-utilities/#insert-annotations-to-source}{\<insert-annotations-to-source>}.
-\item Processor's name.
-\item Target program's classpath.  This argument is required; pass "" if it
-  is empty.
-\item Optional: Extra processor arguments which will be passed to the checker, if any.
-  You may supply any number of such arguments, or none.  Each such argument
-  must start with a hyphen.
-\item Optional: Paths to \<.jaif> files used as input in the inference
-    process.
-\item Paths to \<.java> files in the program.
-\end{enumerate}
+Here are requirements to run the scripts:
+
+\begin{itemize}
+\item The project on which inference is run must contain an Ant, Gradle,
+  or Maven buildfile that compiles the project.
+\item \verb|JAVA8_HOME| environment variable must point to a Java 8 JDK.
+\item \verb|JAVA11_HOME| environment variable must point to a Java 11 JDK.
+\item \<CHECKERFRAMEWORK> environment variable must point to a built copy of the Checker Framework.
+\item Other dependencies:
+  ant,
+  awk,
+  curl,
+  git,
+  gradle,
+  mvn,
+  python2.7 (for dljc),
+  wget.
+
+  Python2.7 modules:
+  subprocess32.
+\end{itemize}
+
+
+\subsectionAndLabel{Running whole-program inference on a single project}{wpi-one}
+
+A typical invocation of whole-program inference is
+
+\begin{Verbatim}
+  wpi.sh -d . -- --checker nullness
+\end{Verbatim}
+
+TODO: What is the result?  A log file?  Placed where?
+
+The full syntax for invoking \<wpi.sh> is
 
-% TODO: Change the example project that is being annotated, since plume-lib is now deprecated.
-For example, to add annotations to the \<plume-lib> project:
 \begin{Verbatim}
-git clone https://github.com/mernst/plume-lib.git
-cd plume-lib
-make jar
-$CHECKERFRAMEWORK/checker/bin/infer-and-annotate.sh \
-    "LockChecker,NullnessChecker" java/plume.jar:java/lib/junit-4.12.jar:$JAVA_HOME/lib/tools.jar \
-    `find java/src/plume/ -name "*.java"`
-# View the results
-git diff
+  wpi.sh -d PROJECTDIR [-t TIMEOUT] -- [DLJC-ARGS]
+\end{Verbatim}
+
+Arguments in square brackets are optional.
+Here is an explanation of the arguments:
+
+\begin{description}
+\item[-d PROJECTDIR]
+  The top-level directory of the project.  It must contain an Ant, Gradle,
+  or Maven buildfile.
+
+\item[-t TIMEOUT]
+  The timeout for running the checker, in seconds
+
+\item[DLJC-ARGS]
+  Arguments that are passed directly to
+  \ahref{https://github.com/kelloggm/do-like-javac}{do-like-javac}'s
+  \<dljc> program without
+  modification.  One argument is required: \<--checker>, which indicates
```

Comment:
The `dljc` usage string does not include `--checker` as a permitted command-line option.